test/evergreen.yml

```diff
@@ -347,15 +348,57 @@ functions:
         ${test_env_vars|} $CTEST -L check ${smp_command|} --output-on-failure ${check_args|} 2>&1
 
   "cppsuite test":
-    command: shell.exec
-    params:
-      # The tests need to be executed in the cppsuite directory as some required libraries have
-      # their paths defined relative to this directory.
-      working_dir: "wiredtiger/cmake_build/test/cppsuite"
-      script: |
-        set -o errexit
-        set -o verbose
-        ${test_env_vars|} ./run -t ${test_name} -C '${test_config}' -f ${test_config_filename} -l 2
+    - command: shell.exec
+      params:
+        # The tests need to be executed in the cppsuite directory as some required libraries have
+        # their paths defined relative to this directory.
+        # The below script saves the exit code from the test to use it later in this function. By
+        # doing this we can define our own custom artifact upload task without it being cancelled by
+        # the test failing.
+        # Additionally if the test fails perf statistics won't be uploaded as they may be invalid
+        # due to the test failure.
+        working_dir: "wiredtiger/cmake_build/test/cppsuite"
+        shell: bash
+        script: |
+          set -o verbose
```

Comment:
Guess it's intentional not to put `set -o errexit` here as you want perf and artifacts uploads to run through in the case of test failures.

Reply:
Yeah set -o errexit breaks this script.